app/src/main/java/com/realityexpander/tasky/agenda_feature/presentation/agenda_screen/AgendaScreen.kt

```diff
@@ -146,6 +155,47 @@ fun AgendaScreenContent(
         (context as MainActivity).exitApp()
     }
 
+    // Handle one-time "sequenced" events (is this a good idea?)
+    LaunchedEffect(state.oneTimeEvent) {
+        state.oneTimeEvent?.let { oneTimeEvent->
+            when (oneTimeEvent) {
+                is AgendaEvent.OneTimeEvent.ScrollToItem -> {
+                    val item = agendaItems.indexOfFirst { it.id == oneTimeEvent.agendaItemId }
+                    if (item >= 0) {
+                        scope.launch {
+                            scrollState.animateScrollToItem(item)
+                            onAction(AgendaEvent.OneTimeEvent.Reset)
+                        }
+                    }
+                }
+                else -> {}
+            }
+        }
+    }
```

Comment:
Not a good idea, since it's very vulnerable to race conditions. If event A hasn't been processed yet and event B is assigned, this whole `LaunchedEffect` block will be cancelled and relaunched. 
Also, you currently don't have a way to reset the one time event. Imagine you show an error here and when a config change happens, this would be re-triggered and show the error again, so there's not really anything one time about this.

Reply:
On Line 167, I am sending a "reset" `OneTimeEvent.Reset` here, only after the "scrolling" is completed.  I have an `onEvent` handler in `AgendaViewModel` to reset the one-time event. Maybe you missed that?

The whole reason I do it this way is to prevent a race condition that was happening when i attempted to use a one-time event using a SharedFlow. 

What I'm trying to do is to scroll to the new item AFTER it has been added to the list. And since the block inside the launched effect is captured as a closure, I dont have access to the current value of `agendaItems` until AFTER it gets updated. And the race condition was occurring when the `agendaItems` is updated when i was using a `SharedFlow` and the `ScrollTo` event was lost because it was cancelled. 

I want to make sure I only trigger the `scrollTo` AFTER the `agendaItems` has been updated. It has to be sequential in this case. This particular one-time event is depending on the data being available from the `agendaState` `stateIn` in the ViewModel. The `ScrollTo` needs to have the current `AgendaItems` available to be able to do a search for the `agendaItem` id. Due to this, only after the data in `agendaItems` has been updated can I perform the "ScrollTo".

So this is not like showing a "snackbar" or "errorMessage". I will have a separate one-time event handler for that using `Channels` or `SharedFlow`. 

I hope this make sense.

Reply:
Ah I see regarding the reset mechanism, I missed that. In your case it's still a problem since the reset action is fired AFTER the scroll animation which means if during the scroll another event comes in, the reset action of the scroll will reset the new event. 
It also still comes with the problem that there can be only one one-time event at a time which is quite against the concept of concurrency. If you really want to use these one-time events I'd just stick to Channels and collect them in `LaunchedEffect(Unit)`. 
If you want to do with with states, you need to make each one-time event a separate state so you allow that there can be multiple at a time. 

Regarding your issue (hope I get you right): Why not do something like this which is executed every time `agendaItems` changes. The logic for `shouldScrollToItem` could be a state or a function of your ViewModel:
```
LaunchedEffect(state.agendaItems) {
    if(shouldScrollToItem) {
        // Scroll to item
    }
}
```

Reply:
OK, I see your code example is a better way to do it. Let me implement that solution and check it out.